tests/test_telemetry.py

```diff
@@ -598,146 +580,83 @@ class Test_TelemetryConfigurationChaining:
-
-        if len(telemetry_data) == 0 and not success_by_default:
-            raise ValueError("No telemetry data to validate on")
-
-        for data in telemetry_data:
-            validator(data)
-
-    def _validate_precedence_order(self, chain: list) -> None:
-        """Validate that a configuration chain follows the official precedence order.
-
-        Args:
-            chain: List of configuration items with 'origin' keys.
-
-        NOTE: This method is used to check that the origins in a configuration chain are in the correct
-        precedence order. In the context of this test, the actual order in the payload is not important;
-        only the relationship between origin precedence and seq_id is validated in the main test logic.
-
-        """
-        precedence_map = {origin: i for i, origin in enumerate(self._ORIGIN_PRECEDENCE_ORDER)}
-
-        for i in range(len(chain) - 1):
-            current_origin = chain[i]["origin"]
-            next_origin = chain[i + 1]["origin"]
-
-            current_precedence = precedence_map.get(current_origin, -1)
-            next_precedence = precedence_map.get(next_origin, -1)
-
-            assert current_precedence < next_precedence, (
-                f"Configuration precedence order violation: '{current_origin}' "
-                f"(precedence {current_precedence}) should come before '{next_origin}' "
-                f"(precedence {next_precedence}) according to {self._ORIGIN_PRECEDENCE_ORDER}"
-            )
-
-    def _validate_configuration_chains(
-        self, configurations: list, expected_config: dict, *, require_seq_id: bool
-    ) -> None:
-        """Validate individual configuration chains against expected values
-
-        Args:
-            configurations: Actual configurations from telemetry data
-            expected_config: Expected configuration structure
-            require_seq_id: Whether to validate seq_id ordering
+    @scenarios.telemetry_app_started_config_chaining
+    def test_app_started_config_chaining_origin_precedence(self):
+        """Assert that the seq_id for each configuration entry for a given configuration name matches the origin precedence"""
 
-        """
-        for cnf_name, expected_chain in expected_config.items():
-            # Filter actual entries for this configuration name
-            actual_chain = [cnf for cnf in configurations if cnf["name"] == cnf_name]
+        def validator(data):
+            # Create array of configs that match the config name
+            matching_configs = []
+            config_name = list(self.CONFIG_PRECEDENCE_ORDER[context.library.name]["configuration"].keys())[0]
+            config_precedence_order = self.CONFIG_PRECEDENCE_ORDER[context.library.name]["configuration"][config_name]
+            for telemetry_payload in data:
+                if get_request_type(telemetry_payload) not in ["app-started", "app-client-configuration-change"]:
```

Comment:
Similar comment to previous -- is this test supposed to include app-client-configuration-change?